packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -760,38 +819,138 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
         this.callAdapter.off('isSpokenLanguageChanged', listener);
         break;
       case 'messageReceived':
-        this.chatAdapter.off('messageReceived', listener);
+        this.chatAdapterPromise.then((adapter) => {
+          adapter.off('messageReceived', listener);
+        });
         break;
       case 'messageEdited':
-        this.chatAdapter.off('messageEdited', listener);
+        this.chatAdapterPromise.then((adapter) => {
+          adapter.off('messageEdited', listener);
+        });
         break;
       case 'messageDeleted':
-        this.chatAdapter.off('messageDeleted', listener);
+        this.chatAdapterPromise.then((adapter) => {
+          adapter.off('messageDeleted', listener);
+        });
         break;
       case 'messageSent':
-        this.chatAdapter.off('messageSent', listener);
+        this.chatAdapterPromise.then((adapter) => {
+          adapter.off('messageSent', listener);
+        });
         break;
       case 'messageRead':
-        this.chatAdapter.off('messageRead', listener);
+        this.chatAdapterPromise.then((adapter) => {
+          adapter.off('messageRead', listener);
+        });
         break;
       case 'chatParticipantsAdded':
-        this.chatAdapter.off('participantsAdded', listener);
+        this.chatAdapterPromise.then((adapter) => {
+          adapter.off('participantsAdded', listener);
+        });
         break;
       case 'chatParticipantsRemoved':
-        this.chatAdapter.off('participantsRemoved', listener);
+        this.chatAdapterPromise.then((adapter) => {
+          adapter.off('participantsRemoved', listener);
+        });
         break;
       case 'callError':
         this.callAdapter.off('error', listener);
         break;
       case 'chatError':
-        this.chatAdapter.off('error', listener);
+        this.chatAdapterPromise.then((adapter) => {
+          adapter.off('error', listener);
+        });
         break;
       default:
         throw `Unknown AzureCommunicationCallWithChatAdapter Event: ${event}`;
     }
   }
 }
 
+/**
+ * Provides a way to get the chat thread ID for a given locator.
+ *
+ * @public
```

Comment:
This is not a public function, make this `@private` if its just used within `react-composites` dir or `@internal` if used by other packlets in this repo.
`@public` is just used if an external developer would use this and it would show up in the public API

Reply:
Same with functions below